Mathlib/Init/Algebra/Order.lean

```diff
@@ -402,15 +402,16 @@ theorem compare_iff (a b : α) {o : Ordering} : compare a b = o ↔ o.toRel a b
   · exact compare_eq_iff_eq
   · exact compare_gt_iff_gt
 
-theorem compare_symm (a b : α) : Ordering.swap (compare a b) = compare b a := by
+-- These are `private` because the user-facing lemmas are the fields of the `TransCmp` instance.
```

Comment:
Why not just prove them inline?

Reply:
I was actually going to ask on zulip 1) if this kind of thing matters (re: expression size and re: time it takes to construct each "new" instance (do these function as autoparams if they're inline, in which case they're run each time?)) and 2) if there's a preferred style, in general.

Reply:
Proofs in `instance`s are only "run" once, just like any other lemma.

Reply:
Ok, sounds good; I'll inline them.

Reply:
A lot of these theorems seem to be duplicated with `TransCmp`, see https://github.com/leanprover/std4/blob/main/Std/Classes/Order.lean . You should just prove or assert that the `Ord` is lawful and get the theorems that way.

Reply:
Which ones are duplicated? All of the new ones relate `Ordering` to `<`, `=`, and `>`, which as far as I can see doesn't happen in Std.Classes.Order